ceno_zkvm/src/bin/e2e.rs

```diff
@@ -131,14 +135,77 @@ fn main() {
     );
 
     let max_steps = args.max_steps.unwrap_or(usize::MAX);
-    run_e2e(
-        program,
-        platform,
-        args.stack_size,
-        args.heap_size,
-        hints,
-        max_steps,
+
+    type E = GoldilocksExt2;
+    type Pcs = Basefold<GoldilocksExt2, BasefoldRSParams>;
+
+    let (prover, verifier, zkvm_witness, pi, cycle_num, e2e_start, exit_code) =
+        run_e2e_gen_witness::<E, Pcs>(
+            program,
+            platform,
+            args.stack_size,
+            args.heap_size,
+            hints,
+            max_steps,
+        );
+
+    let timer = Instant::now();
+    let mut zkvm_proof = run_e2e_proof(prover, zkvm_witness, pi);
+    let proving_time = timer.elapsed().as_secs_f64();
+    let e2e_time = e2e_start.elapsed().as_secs_f64();
+    let witgen_time = e2e_time - proving_time;
+    println!(
+        "Proving finished.\n\
```

Comment:
You can just use 'natural' newlines?

```suggestion
        "Proving finished.
```